oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/MissingLastRevSeekerTest.java

```diff
@@ -216,4 +220,26 @@ public void getClusterNodeInfo() {
     private ClusterNodeInfoDocument getClusterNodeInfo(int clusterId) {
         return seeker.getClusterNodeInfo(clusterId);
     }
+    
+    @Test
+    public void getNonSplitDocs() throws Exception {
+        String nodeName = this.getClass().getName() + "-foo";
+        DocumentNodeStore dns = getBuilder().setDocumentStore(store).getNodeStore();
```

Comment:
The DocumentNodeStore should use the clock setup by this test. Otherwise it can be unstable because `Revision` and `ClusterNodeInfo` are using the virtual clock. Unfortunately it is also necessary to wrap the store and prevent disposal of the underlying store.
```suggestion
        DocumentNodeStore dns = getBuilder().clock(clock).setAsyncDelay(0)
                .setDocumentStore(new DocumentStoreWrapper(store) {
                    @Override
                    public void dispose() {
                        // do not close underlying store, otherwise cleanup
                        // cannot remove documents after the test
                    }
                }).getNodeStore();
```

Reply:
Does this wrapping is required if we are not disposing the nodestore?

Reply:
Disposal is needed, otherwise some background threads continue to run and occupy a clusterId. E.g. the lease update thread.